src/coreComponents/codingUtilities/Section.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Section.hpp
+ */
+
+#ifndef GEOS_COMMON_SECTION_HPP
+#define GEOS_COMMON_SECTION_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class Section
+{
+public:
+
+  Section();
+
+  /**
+   * @brief Set the name of the section
+   * @param m_sectionTitle The name of the section
+   */
+  void setName( string_view m_sectionTitle );
+
+  /**
+   * @brief Add a description to the section by concatening a description name and descriptions values.
+   * @param descriptionName The description name
+   * @param args Descriptions values to be aligned.
+   * Descriptions values can be be any types and will be aligned
+   */
+  template< typename ... Args >
+  void addDescription( string const & descriptionName, Args const & ... args );
```

Comment:
Currently this works, but if someone want to show the description at the `end()`, it will have the same strange behaviour I mentionned at `setName()`.
You can document `end()` implementation to never show non-const attributes, or set the description as constant too.

Reply:
setName removed